### ISSUE TITLE
Result as Json

### ISSUE THREAD

**Issue by gitorrin** [jerriep/dotnet-outdated#57]
Highlighting outdated references in the command prompt is great. 

Would it be possible to return a Json output, as a result of the analysis? 
That'd make it easier to parse and create additional logic based on the report content.

Thanks

**Comment by patros** [comment]
@jerriep have you given any thought to the json schema for this? I'd be happy to have a go at implementing this one if you haven't started already.

**Comment by jerriep** [comment]
@patros No, I have not thought about it yet. I can give attention to it probably around the weekend, but you're also welcome to make a suggestion in the meantime (if you have time on your hands)

**Comment by patros** [comment]
Although that ^ doesn't indicate the semantic change. i.e. major, minor or patch. So need something added there. Maybe move the logic from `GetUpgradeSeverityColor` to the `Dependency` object and make it a property there?

**Comment by jerriep** [comment]
Yeah, you can create an `UpgradeSeverity` enum and add a property for that to `Dependency`:

```
public enum UpgradeSeverity
{
    Patch,
    Minor,
    Major
}
```

Regarding the XML, yeah that is fine. For each project you will then have the Target Framework(s) listed and underneath that the list of outdated dependencies with their properties. For the `projects` node in your suggested scema, you can basically just serialize that `List<Project>` after `AnalyzeDependencies` is called.  

But make sure you exclude some of the properties from serialization, such as `Description` and `VersionRange`. Also, for properties such as `ResolvedVersion` and `LatestVersion`, you'll need to serialize them to a simple string value - by default they will be serialized with all of the child properties, and we don't want that.

**Comment by patros** [comment]
Cool, sounds like a plan. I'll try and knock this off over the next couple of days.

**Comment by jerriep** [comment]
Closes by #100
